src/client/components/CodeDiffView/CodeDiffView.js

```diff
@@ -21,22 +22,33 @@ export type Props = {
   onDeleteInlineComment: (commentId: string, text: string) => void,
 }
 
+const defaultEmptyList = []
+
 class CodeDiffView extends PureComponent {
+  static defaultProps = {
+    viewType: '0',
+  }
+
   props: Props
 
+  shouldComponentUpdate(nextProps: Object, nextState: Object) {
+    return (this.props.rawDiff || this.props.unifiedDiff || this.props.sideBySideDiff) &&
```

Comment:
The component should update if
1. the current props contain at least of the kinds of diffs, _and_
2. the next props are different from the current.

Is 1. correct? Shouldn't it be if the _next_ props contain a diff?